### PR TITLE
First try at month/year menus

### PR DESCRIPTION
**Pull request by rodryquintero**
what do you think?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,5 +87,8 @@
     "react": ">=0.14",
     "react-dom": ">=0.14",
     "react-addons-shallow-compare": ">=0.14"
+  },
+  "jshintConfig": {
```
Comment (ljharb) [inline package.json]: please remove this; we don't use jshint.

File: src/components/DateRangePicker.jsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ import {
   START_DATE,
   END_DATE,
   HORIZONTAL_ORIENTATION,
-  VERTICAL_ORIENTATION,
+  VERTICAL_ORIENTATION
```
Comment (ljharb) [inline src/components/DateRangePicker.jsx]: trailing commas are required by our linter everywhere they are supported.

**Comment by ljharb** [comment]
Can you include a screenshot and/or a link to a demo page?

**Comment by rodryquintero** [comment]
Sure. Here are some.
image

image
image

**Comment by ljharb** [comment]
What happens if you select a month/year that falls outside of the selectable range?

**Comment by rodryquintero** [comment]
It displays the calendar but the days are grayed out. You cannot select them.

image

This, of course, is not ready to be merged as all tests have failed. I do not have experience using the test tools so I did not check before doing the pull request.

Feel free to delete this pull request. Just wanted to do a "proof of concept" and show you.

**Comment by ljharb** [comment]
It seems like if you can navigate to a useless month, there should be an easy way to jump back to the closest month with selectable days. Alternatively, based on the "year" setting, months with no selectable days should not be selectable.

**Comment by rodryquintero** [comment]
Will get to it. Thanks for the help.

**Comment by arnars** [comment]
How is this coming along? We could really use this feature in our current project.

**Comment by selvagsz** [comment]
Any progress in this feature ?

**Comment by majapw** [comment]
This particular implementation is going to be really hard to resolve with the structural changes that have been included in react-dates since this PR was opened. I'm going to be investigating this conceptually pretty soon tho and trying to push something through the door.

**Pull request closed without merging**